enigma-principal/enclave/src/epoch_keeper_t/nested_encoding.rs

```diff
@@ -0,0 +1,58 @@
+use std::vec::Vec;
+use ethereum_types::{U256,H160};
+
+pub const ONE: u8 = 1;
+pub const ZERO: u8 = 0;
+/// implements the serialization for types needed for epoch encoding in the KM node,
+/// according to this proof: https://github.com/enigmampc/protocol-discovery/blob/master/docs/hash_mul_nested.pdf
+pub trait NestedSerialization {
+    fn hash_encode(&self) -> (usize, Vec<u8>);
+}
+
+impl NestedSerialization for U256 {
+    fn hash_encode(&self) -> (usize, Vec<u8>) {
+        let mut res: Vec<u8> = Vec::new();
+        let mut msg = [0u8;32];
+
+        self.to_big_endian(&mut msg);
+        let len = msg.len().to_be_bytes();
+
+        res.push(ZERO);
+        res.extend_from_slice(&len);
+        res.extend_from_slice(&msg);
+        (msg.len(), res)
```

Comment:
According to the proof doc the returned length should be the length of the entire serialization, which is  `res.len()`